reward/reward_distributor_test.go

```diff
@@ -81,246 +188,998 @@ func Test_isEmptyAddress(t *testing.T) {
+			isKore:  false,
+			isMagma: true,
+			fee:     1000,
+			expected: &Result{
+				total:  1000,
+				reward: 500,
+				burnt:  500,
+			},
+		},
+		{
+			desc:    "isKore=false, isMagma=true, fee=10e18 [011]",
+			isKore:  false,
+			isMagma: true,
+			fee:     10e18,
+			expected: &Result{
+				total:  10e18,
+				reward: 5e18,
+				burnt:  5e18,
+			},
+		},
+		{
+			desc:    "isKore=true, isMagma=true, fee=1000 [110]",
+			isKore:  true,
+			isMagma: true,
+			fee:     1000,
+			expected: &Result{
+				total:  1000,
+				reward: 0,
+				burnt:  1000,
+			},
+		},
+		{
+			desc:    "isKore=true, isMagma=true, fee=10e18 [111]",
+			isKore:  true,
+			isMagma: true,
+			fee:     10e18,
+			expected: &Result{
+				total:  10e18,
+				reward: 4.3472e18, // 5 - minted*0.34*0.2
+				burnt:  5.6528e18, // 5 + minted*0.34*0.8
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		header := &types.Header{
+			Number:     big.NewInt(1),
+			GasUsed:    tc.fee,
+			BaseFee:    big.NewInt(1),
+			Rewardbase: proposerAddr,
+		}
+
+		config := getTestConfig()
+		if !tc.isKore {
+			config = noKore(config)
+		}
+		if !tc.isMagma {
+			config = noMagma(config)
+		}
+
+		rc, err := NewRewardConfig(header, config)
+		assert.Nil(t, err)
+
+		total, reward, burnt := calcDeferredFee(rc)
+		actual := &Result{
+			total:  total.Uint64(),
+			reward: reward.Uint64(),
+			burnt:  burnt.Uint64(),
+		}
+		assert.Equal(t, tc.expected, actual, "failed tc: %s", tc.desc)
+	}
+}
+
+func TestRewardDistributor_calcDeferredFee_nodeferred(t *testing.T) {
+	header := &types.Header{
+		Number:     big.NewInt(1),
+		GasUsed:    1000,
+		BaseFee:    big.NewInt(1),
+		Rewardbase: proposerAddr,
+	}
+
+	rc, err := NewRewardConfig(header, noDeferred(getTestConfig()))
+	assert.Nil(t, err)
+
+	total, reward, burnt := calcDeferredFee(rc)
+	assert.Equal(t, uint64(0), total.Uint64())
+	assert.Equal(t, uint64(0), reward.Uint64())
+	assert.Equal(t, uint64(0), burnt.Uint64())
+}
+
+func TestRewardDistributor_calcSplit(t *testing.T) {
+	type Result struct{ proposer, stakers, kgf, kir, remaining uint64 }
+
+	header := &types.Header{
+		Number:  big.NewInt(1),
+		BaseFee: big.NewInt(0), // placeholder
+	}
+
+	testcases := []struct {
+		desc     string
+		isKore   bool
+		fee      uint64
+		expected *Result
+	}{
+		{
+			desc:   "kore=false, fee=0",
+			isKore: false,
+			fee:    0,
+			expected: &Result{
+				proposer:  3.264e18, // minted * 0.34
+				stakers:   0,
+				kgf:       5.184e18, // minted * 0.54
+				kir:       1.152e18, // minted * 0.12
+				remaining: 0,
+			},
+		},
+		{
+			desc:   "kore=false, fee=55555",
+			isKore: false,
+			fee:    55555,
+			expected: &Result{
+				proposer:  3.264e18 + 18888, // (minted + fee) * 0.34
+				stakers:   0,
+				kgf:       5.184e18 + 29999, // (minted + fee) * 0.54
+				kir:       1.152e18 + 6666,  // (minted + fee) * 0.12
+				remaining: 2,
+			},
+		},
+		{
+			desc:   "kore=true, fee=0",
```

Comment:
I guess the reward distribution has changed due to KGP-3:
https://govforum.klaytn.foundation/t/kgp-3-reduction-of-klaytn-block-reward/117
Please double check if the tests are not affected by the recent change of reward distribution.

Reply:
let's address this in another PR